### PR TITLE
Cleanup redundant transitive dependency usage/declarations

### PR DESCRIPTION
**Pull request by joakime**
Revert the `maven-enforcer-plugin` to `3.0.0-M3`

Disabled `<requireUpperBoundDeps>` configuration.

Eliminate redundant transitive dependency declarations.

### CODE REVIEW

**Review by janbartel — CHANGES_REQUESTED** [review]
I am definitely in favour of reverting to `3.0.0-M3` for the `enforcer` plugin.

I don't think we need to remove the `enforceUpperBound` rule, _unless_ that too was added at the same time as the upgrade to `3.0.0`.

I'm not convinced about adding in all these other changes because a) some of them seem to go against the convention of moving every dependency into the `dependencyManagement` section of the pom that I thought we were supposed to be striving for and b) the build of this branch doesn't work and it will be difficult to work out why.

I think we should just do a simple PR to change back the version of the `enforcer` plugin, so we are back to a known working configuration,  then do a subsequent PR with a cleanup of the dependencies.

File: jetty-jspc-maven-plugin/pom.xml [hunk 1]
```diff
@@ -76,19 +76,23 @@
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
+      <version>${maven.version}</version>
```
Comment (olamy) [inline jetty-jspc-maven-plugin/pom.xml:79]: no need of this as it's already declared in dptMngt in top level

File: jetty-maven-plugin/pom.xml [hunk 2]
```diff
@@ -106,27 +106,32 @@
     <dependency>
       <groupId>org.apache.maven.shared</groupId>
       <artifactId>maven-artifact-transfer</artifactId>
-      <version>0.13.1</version>
+      <version>${maven-artifact-transfer.version}</version>
     </dependency>
     <dependency>
       <groupId>org.apache.maven</groupId>
       <artifactId>maven-plugin-api</artifactId>
+      <version>${maven.version}</version>
```
Comment (olamy) [inline jetty-maven-plugin/pom.xml:114]: not needed

File: pom.xml [hunk 3]
```diff
@@ -846,26 +849,11 @@
         <artifactId>javax.annotation-api</artifactId>
         <version>${javax.annotation.api.version}</version>
       </dependency>
-      <dependency>
-        <groupId>org.ow2.asm</groupId>
-        <artifactId>asm</artifactId>
-        <version>${asm.version}</version>
```
Comment (olamy) [inline pom.xml:852]: we use it directly and having this here in dptMngt will insure coherence between modules especially for transitive dependencies as it;s the role of dependencyManagement section.
See
https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management

```
A second, and very important use of the dependency management section is 
to control the versions of artifacts used in transitive dependencies.
```

File: pom.xml [hunk 4]
```diff
@@ -911,53 +899,7 @@
         <version>${javax.transaction.api.version}</version>
         <scope>provided</scope>
       </dependency>
-      <!-- maven deps -->
```
Comment (olamy) [inline pom.xml:914]: we use it directly and having this here in dptMngt will insure coherence between modules

**Review by olamy — CHANGES_REQUESTED** [review]

File: jetty-unixsocket/pom.xml [hunk 5]
```diff
@@ -11,6 +11,7 @@
   <properties>
     <bundle-symbolic-name>${project.groupId}.unixsocket</bundle-symbolic-name>
     <spotbugs.onlyAnalyze>org.eclipse.jetty.unixsocket.*</spotbugs.onlyAnalyze>
+    <jnr-unixsocket.version>0.38.10</jnr-unixsocket.version>
```
Comment (olamy) [inline jetty-unixsocket/pom.xml:14]: I would prefer having this in single place (i.e top pom)

File: pom.xml [hunk 6]
```diff
@@ -91,7 +91,8 @@
     <maven.compiler.plugin.version>3.8.1</maven.compiler.plugin.version>
     <maven.dependency.plugin.version>3.2.0</maven.dependency.plugin.version>
     <maven.eclipse.plugin.version>2.10</maven.eclipse.plugin.version>
-    <maven.enforcer.plugin.version>3.0.0</maven.enforcer.plugin.version>
+    <maven.enforcer.plugin.version>3.0.0-M3</maven.enforcer.plugin.version>
```
Comment (olamy) [inline pom.xml:94]: :(

File: pom.xml [hunk 7]
```diff
@@ -220,11 +223,11 @@
                 <versionOsgiRule implementation="org.eclipse.jetty.toolchain.enforcer.rules.RequireOsgiCompatibleVersionRule" />
                 <versionRedhatRule implementation="org.eclipse.jetty.toolchain.enforcer.rules.RequireRedhatCompatibleVersionRule" />
                 <versionDebianRule implementation="org.eclipse.jetty.toolchain.enforcer.rules.RequireDebianCompatibleVersionRule" />
-                <requireUpperBoundDeps>
+                <!--<requireUpperBoundDeps>
```
Comment (olamy) [inline pom.xml:226]: :(

File: pom.xml [hunk 8]
```diff
@@ -1274,6 +1150,7 @@
       <properties>
         <settingsPath>${env.GLOBAL_MVN_SETTINGS}</settingsPath>
         <surefire.rerunFailingTestsCount>3</surefire.rerunFailingTestsCount>
+        <spotbugs.plugin.version>4.3.0</spotbugs.plugin.version>
```
Comment (olamy) [inline pom.xml:1153]: I would prefer having this in single place (top of the pom)

File: tests/test-sessions/test-infinispan-sessions/pom.xml [hunk 9]
```diff
@@ -103,42 +103,6 @@
...
-      <version>${infinispan.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
-      <groupId>org.infinispan</groupId>
-      <artifactId>infinispan-client-hotrod</artifactId>
-      <version>${infinispan.version}</version>
-      <scope>test</scope>
-    </dependency>
-    <dependency>
```
Comment (olamy) [inline tests/test-sessions/test-infinispan-sessions/pom.xml:124]: some part of the code use so must be declare

**Review by olamy — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 4 above)
Comment (olamy) [reply pom.xml:914]: part of documentation here https://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html#Dependency_Management 
```
The dependency management section is a mechanism for centralizing dependency information. 
When you have a set of projects that inherit from a common parent, 
it's possible to put all information about the dependency in the common POM and have simpler references 
to the artifacts in the child POMs.
```

**Pull request closed without merging**